index.js

```diff
@@ -1 +1,97 @@
 // implement your API here
+const express = require('express') //import express
+
+const Users = require("./data/db.js") //import server
+
+const server = express(); //start server
+
+server.use(express.json()); //tells server to parse JSON
+
+//GET function returns all users.
+server.get("/users", (req, res) => {
+    Users.find()
+        .then(user => {
+            res.status(200).json(user);
+        })
+        .catch(err => {
+            res.status(500).json({ message: "The users information could not be retrieved." })
+        })
+})
+
+//GET function returns specific ID
+server.get("/users/:id", (req, res) => {
+    const { id } = req.params;
+
+    Users.findById(id)
+        .then(user => {
+            if (id) {
+                res.status(200).json(user)
+            } else {
+                res.status(404).json({ message: "User not found" })
+            }
+        })
+        .catch(err => {
+            res.status(500).json({ error: "The user information could not be retrieved." })
+        })
+})
+
+//POST function to add a new user 
+server.post('/users', (req, res) => {
+    const userInfo = req.body;
+    console.log(userInfo);
+
+    Users.insert(userInfo)
+        .then(person => {
+            if (userInfo.name && userInfo.bio) {
+                Users.findById(person.id)
+                    .then(user => {
+                        res.status(200).json(user)
+                    })
+            } else {
+                res.status(400).json({ message: "Name and Bio are required" })
+            }
+        })
+        .catch(err => {
+            res.status(500).json(err);
+        });
+});
+
+//DELETE function to delete a user
+server.delete("/users/:id", (req, res) => {
+    const { id } = req.params
+
+    Users.remove(id)
+        .then(user => {
+            if (user) {
+                res.status(204).end()
```

Comment:
They don't specify what to return from a successful delete, but we should probably return something so the user knows it has been successful. Status 204 means no content is returned, but there might be useful information on the headers. Check out this link: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204 . So you could maybe do something with that if you would like, but it's not something I would be familiar with right away. Instead, I like to return a 200 status code, and I'll send back the list of the new users. Do this by doing another Users.find(), and return 200
```suggestion
                Users.find().then(data => res.status(200).json(data));
```